src/main/java/zmaster587/advancedRocketry/tile/TileGuidanceComputer.java

```diff
@@ -137,41 +147,73 @@ else if(itemType instanceof ItemSatelliteIdentificationChip) {
-			if(landingDimension == Configuration.spaceDimId) {
-				//TODO: handle Exception
-				ISpaceObject object = SpaceObjectManager.getSpaceManager().getSpaceStation(ItemStationChip.getUUID(stack));
-				HashedBlockPosition vec = null;
-				if(object instanceof SpaceObject) {
-					if(landingLoc.get(object.getId()) != null) {
-						vec = landingLoc.get(object.getId());
-
-						if(commit)
-							((SpaceObject)object).getPadAtLocation(landingLoc.get(object.getId())).setOccupied(true);
+		//ToDo: replace all nulls with current coordinates of the ship.
+		//Make the if tree match the destination if tree:
+		if(!stack.isEmpty()){
+			Item itemType = stack.getItem();
+			if (itemType instanceof ItemPlanetIdentificationChip) {
+				//This could be the location of the rocket.
+				return null;
+			}
+			else if(itemType instanceof ItemStationChip) {
+				ItemStationChip chip = (ItemStationChip)stack.getItem();
+				if(landingDimension == Configuration.spaceDimId) {
+					//TODO: handle Exception
+					ISpaceObject object = SpaceObjectManager.getSpaceManager().getSpaceStation(ItemStationChip.getUUID(stack));
+					HashedBlockPosition vec = null;
+					if(object instanceof SpaceObject) {
+						if(landingLoc.get(object.getId()) != null) {
+							vec = landingLoc.get(object.getId());
+
+							if(commit)
+								((SpaceObject)object).getPadAtLocation(landingLoc.get(object.getId())).setOccupied(true);
+						}
+						else
+							vec = ((SpaceObject)object).getNextLandingPad(commit);
 					}
-					else
-						vec = ((SpaceObject)object).getNextLandingPad(commit);
-				}
 
-				if(object == null)
-					return null;
+					if(object == null)
+						return null;
 
-				if(vec == null)
-					vec = object.getSpawnLocation();
+					if(vec == null)
+						vec = object.getSpawnLocation();
 
-				return new Vector3F<Float>(new Float(vec.x), new Float(vec.y), new Float(vec.z));
-			}
-			else {
-				LandingLocation loc = chip.getTakeoffCoords(stack, landingDimension);
-				if(loc != null)
-				{
-					return loc.location;
+					return new Vector3F<Float>(new Float(vec.x), new Float(vec.y), new Float(vec.z));
+				}
+				else {
+					LandingLocation loc = chip.getTakeoffCoords(stack, landingDimension);
+					if(loc != null)
+					{
+						return loc.location;
+					}
+					return null;
 				}
+			}
+			else if(itemType instanceof ItemAsteroidChip) {
+				//Caution Side-Effect dependency: landingPos from getDim.				
+				return landingPos;
+			}
+			else if(itemType instanceof ItemSatelliteIdentificationChip) {
+				//You can't actually go to the Satellites
 				return null;
+			} 
+			else if (stack.getItem() == LibVulpesItems.itemLinker && ItemLinker.getDimId(stack) != Constants.INVALID_PLANET)
+			{
+				//Use the destination the Linker in the Guidance computer directs.
+				BlockPos landingBlock = ItemLinker.getMasterCoords(stack);
+				return new Vector3F<Float>(landingBlock.getX() + 0.5f, (float)Configuration.orbit, landingBlock.getZ() + 0.5f);
```

Comment:
```suggestion
				return new Vector3F<Float>(landingBlock.getX(), (float)Configuration.orbit, landingBlock.getZ());
```

We probably don't need the 0.5f here.  The rocket needs to handle its offset based on its size.  A rocket that's an odd number of blocks by an odd number of blocks in size will probably benefit from the 0.5 offset, but a rocket with an even number of blocks will now by off by 0.5 from where it should be.

Ultimately it probably needs to be solved by returning ints instead of floats and letting the rocket handle its own offsets since the guidance computer has no knowledge of the rocket's size or shape.  But that's out of scope here anyway I think.  For now I think just removing the 0.5f is good.

Reply:
I got the 0.5f bits from the existing code for the docking pads when the linker is in it.  I was being consistent with that.

Reply:
Example: https://github.com/zmaster587/AdvancedRocketry/blob/52b631e6a4c1cc256af0d7f3f90c34761fef8754/src/main/java/zmaster587/advancedRocketry/tile/station/TileLandingPad.java#L159

Reply:
I just tried an even sized rocket and it landed exactly the same way it looked when it launched.  Offset a bit as the builder had done.
![LinkerInPadLanding](https://user-images.githubusercontent.com/4486117/62000680-3b29c480-b0ab-11e9-83e7-51e474c9c27a.png)
![LinkerLandingInGC](https://user-images.githubusercontent.com/4486117/62000681-3b29c480-b0ab-11e9-9475-86e24f9c05ff.png)